### PR TITLE
Add Model Docs

### PR DESCRIPTION
**Pull request by fwrs**

### CODE REVIEW

File: definitions/enums/relationship_type.yml [hunk 1]
```diff
@@ -1,13 +1,19 @@
 ---
 name: RelationshipType
+description: Different kinds of relationships.
```
Comment (zeyla) [inline definitions/enums/relationship_type.yml]: "The type of a relationship between two users."

File: definitions/structs/embed.yml [hunk 2]
```diff
@@ -1,6 +1,15 @@
...
+description: >
+  Represents a rich embed which allows using richer markdown,
+  multiple columns and more. This was heavily inspired by [slack's
+  attachments](https://api.slack.com/docs/message-attachments).
+
+  You can include an attachment in your own message by a user or a bot,
+  or in a webhook. They can be edited, too.
+
+  **Note**: Maximum amount of characters in a field name is 256,
+  in field value - 1024 and in description - 2048
```
Comment (zeyla) [inline definitions/structs/embed.yml]: "256 in a field name, 1024 in a field value, and 2048 in a description."

Comment (zeyla) [inline definitions/structs/embed.yml]: We should also document this in the description/name/value methods in [CreateEmbed] and [CreateEmbedField].

[CreateEmbed]: https://serenity.zey.moe/serenity/utils/builder/struct.CreateEmbed.html
[CreateEmbedField]: https://serenity.zey.moe/serenity/utils/builder/struct.CreateEmbedField.html

File: definitions/structs/emoji.yml [hunk 3]
```diff
@@ -1,12 +1,17 @@
...
+  work on one server. Maximum amount of Emoji a guild can have
+  is 50.
 fields:
   - name: id
     description: "The Id of the emoji."
     type: EmojiId
   - name: name
-    description: "The name of the emoji."
+    description: "The name of the emoji. It must be at least 2 characters
+      long and can only contain alphanumeric characters and underscores."
```
Comment (zeyla) [inline definitions/structs/emoji.yml]: This should be documented in `Context::create_emoji` as well

File: definitions/structs/game.yml [hunk 4]
```diff
@@ -1,14 +1,24 @@
 ---
 name: Game
-description: Information about a game that a user is playing
+description: >
+  Represents a game that a a user is playing or
+  streaming, in case a twitch stream link is provided.
```
Comment (zeyla) [inline definitions/structs/game.yml]: Let's remove "twitch", there might be other services in the future

File: definitions/structs/game.yml [hunk 5]
```diff
@@ -1,14 +1,24 @@
...
+    description: >
+      Allows to differentiate streaming and playing statuses.
     default: GameType::Playing
     from: type
     type: GameType
   - name: name
+    description: >
+      The name of the game that's being played.
+
+      **Note**: Maximum length is 64.
```
Comment (zeyla) [inline definitions/structs/game.yml]: This seems to be a max length of 128 through my tests

File: definitions/structs/guild.yml [hunk 6]
```diff
@@ -1,61 +1,102 @@
...
+    description: >
+      All custom emojis of a server. Such are made using the API or
+      Twitch integrations.
     custom: decode_emojis
     t: EmojiId, Emoji
     type: hashmap
   - name: features
+    description: >
+      VIP server features a server has. Can be obtained through
+      [Discord Partnership website](https://discordapp.com/partners).
```
Comment (zeyla) [inline definitions/structs/guild.yml]: Let's modify this anchor to:

```yaml
- name: features
  description: >
    VIP server features a guild has. These are features obtained through
    the [Discord Partnership] website.

      [Discord Partnership]: https://discordapp.com/partners

File: definitions/structs/guild.yml [hunk 7]
```diff
@@ -1,61 +1,102 @@
...
   - name: owner_id
+    description: Id of the server's owner.
     type: UserId
   - name: presences
+    description: Presence statuses of members.
     custom: decode_emojis
     t: UserId, Presence
     type: hashmap
   - name: region
+    description: Region the server is located in.
```
Comment (zeyla) [inline definitions/structs/guild.yml]: "The region that the guild's voice servers are located in."

File: definitions/structs/guild.yml [hunk 8]
```diff
@@ -1,61 +1,102 @@
...
     type: string
   - name: roles
+    description: All roles a server has.
     custom: decode_roles
     t: RoleId, Role
     type: hashmap
   - name: splash
+    description: >
+      If InviteSplash feature is enabled, this can point to splash image
+      URL displayed when someone opens invite URL.
```
Comment (zeyla) [inline definitions/structs/guild.yml]: Let's link this to the Feature's variant,

```yaml
- name: splash
  description: >
    If the [InviteSplash] feature is enabled, ...

      [InviteSplash]: enum.Feature.html#variant.InviteSplash
```

File: definitions/structs/message.yml [hunk 9]
```diff
@@ -3,43 +3,63 @@ name: Message
...
     optional: true
     type: string
   - name: embeds
+    description: Array of embeds a message has.
     array: true
     type: Embed
   - name: kind
+    description: >
+      Lets you differentiate system messages and messages sent by
+      real accounts.
```
Comment (zeyla) [inline definitions/structs/message.yml]: Webhooks send MessageType 0, so let's `s/messages sent by real accounts/regular messages`

File: definitions/structs/message.yml [hunk 10]
```diff
@@ -3,43 +3,63 @@ name: Message
...
     type: User
   - name: nonce
+    description: Non-repeating number used for ensuring message order.
     optional: true
     type: string
   - name: pinned
+    description: True if message is pinned.
     type: bool
   - name: reactions
+    description: Array of reactions message received.
```
Comment (zeyla) [inline definitions/structs/message.yml]: "Array of reactions performed on the message."

File: definitions/structs/user.yml [hunk 11]
```diff
@@ -1,17 +1,28 @@
...
     default: 'false'
     type: bool
   - name: discriminator
+    description: >
+      The account's discriminator. The name + discriminator pair is
+      always unique.
     type: string
   - name: name
+    description: >
+      The account's username. Changing username may trigger
```
Comment (zeyla) [inline definitions/structs/user.yml]: `s/may trigger discriminator change/will trigger a discriminator change`

File: definitions/structs/game.yml (hunk 5 above)
Comment (fwrs) [inline definitions/structs/game.yml]: i blame fuyu

**Comment by zeyla** [comment]
Rebased in 2844ae158f3d8297b17a584ff9a75f1f51116f48, thanks.

**Pull request closed without merging**